.github/workflows/nightly-release.yml

```diff
@@ -0,0 +1,25 @@
+name: nightly-release
+
+on:
+  schedule:
+    # 13:00 UTC is 05:00 in Pacific standard time (UTC-8), which is well
+    # after nightly TensorFlow wheels are released (around 2--3 AM).
+    # (cron syntax: minute hour day-of-month month day-of-week)
+    - cron: '0 13 * * *'
+
+jobs:
+  ci:
+    uses: ./.github/workflows/ci.yml
+
+  nightly-release:
+    runs-on: ubuntu-18.04
+    needs: ci
+    if: github.repository == 'tensorflow/tensorboard'
+    steps:
+      - name: Publish the package
+        run: |
+          twine check /tmp/tb_nightly_pip_package/tb_nightly*.whl
```

Comment:
Updated to just use the `download-artifact` action. The temporary filesystem isn't documented well enough for me to justify using it.

Reply:
SGTM, thanks.

One additional thought - it might be worth double-checking the behavior if we have a test failure or fail to push the new artifact properly in the CI workflow, just to ensure that we avoid re-uploading the previously published artifact as a new nightly release?

Reply:
The [needs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds) block ensures that the upstream CI job is run successfully before the dependent upload job is attempted. The only way that we would run the `nightly-release` job if any of the previous steps failed is if we explicitly provided `if: ${{ always() }}`.

Reply:
Yes, I agree it should all work as described; I was just thinking since it's a significant change it might be good to sanity-check that a test failure does actually block the release properly. But up to you.

Reply:
Nice!

Reply:
That looks great, thanks for going the extra mile to confirm this!  And good to know re the upload/download artifact behavior, I didn't realize that, but yes that definitely makes it easier to reason about.